contracts/vaults/VaultPauser/RibbonVaultPauser.sol

```diff
@@ -150,26 +141,30 @@ contract RibbonVaultPauser is
         address currentVaultAddress = address(msg.sender);
         IRibbonThetaVault currentVault = IRibbonThetaVault(currentVaultAddress);
 
-        PauseReceipt[] storage pauseReceipts =
-            pausedPositions[currentVaultAddress][_account];
-
-        // user can pause multiple position hence it's necessary to make an array
-        // to store each paused position
-        pauseReceipts.push(
-            PauseReceipt({
-                round: uint16(currentVault.vaultState().round),
-                account: address(_account),
-                shares: uint104(_amount)
-            })
+        // check if position is paused
+        require(
+            !pausedPositions[currentVaultAddress][_account].paused,
+            "Position is paused"
         );
 
-        emit Pause(
+        // transfer from user to pauser
+        IERC20(currentVaultAddress).safeTransferFrom(
             _account,
-            currentVaultAddress,
-            _amount,
-            currentVault.vaultState().round
+            address(this),
+            _amount
         );
 
+        uint16 round = currentVault.vaultState().round;
+
+        pausedPositions[currentVaultAddress][_account] = PauseReceipt({
+            round: round,
+            account: address(_account),
```

Comment:
We can remove `account` from the PauseReceipt because we're already using `account` to identify the PauseReceipt